packages/fxa-settings/src/components/Settings/index.tsx

```diff
@@ -51,13 +54,16 @@ export const Settings = ({
 
   useEffect(() => {
     function handleWindowFocus() {
+      if (account.uid !== currentAccount()?.uid) {
```

Comment:
The only potential issue I see here is that `setCurrentAccount` operates under the assumption that an account state with the provided UID exists in local storage. If you look at the current usages, the account is typically persisted first, then `setCurrentAccount` is called, so there might be an edge case if local storage is cleared and then tab is refocused.

Reply:
I'll do some testing with this scenario and see how it does. Thanks!

Reply:
If local storage is cleared from another tab by going to `/clear` before the tab is refocused, then this condition doesn't apply. It skips to the `session.isDestroyed` and hard navigates back to the index.